### ISSUE TITLE
Use an external theme without customisation

### ISSUE THREAD

**Issue by quozl** [sugarlabs/www-sugarlabs#305]
Much of the noise and wasted effort on our web site has arisen from using themes or stylesheets that are copied into our site and then customised extensively.  This isn't necessary.  Instead, we should use an upstream, well-known, and maintained theme, without customising it.  This may let us concentrate on content rather than form.

**Comment by amaaniqbal** [comment]
+1, I mentioned about this indirectly in the past. Bootstrap v4.3 would replace many of the obsolete stylesheets we are currently having. Bootstrap is the world's most popular framework to develop responsive websites. 

It's maintained [here](https://github.com/twbs/bootstrap/issues).

**Comment by quozl** [comment]
Thanks.  On the other hand, I don't want to cause more noise than signal.  In past year we have had about four content changes, out of 74 commits.  The larger problem we have is very little content being generated. :grin:

**Comment by amaaniqbal** [comment]
True! Have you thought of any workflow to solve either of theme or content related problems?

By the way which content is actually being referred to here? The one on the SugarLabs website?

**Comment by quozl** [comment]
> Have you thought of any workflow to solve either of theme or content related problems?

I don't understand the question.  It seems very vague.  We don't need a workflow to solve problems.

> By the way which content is actually being referred to here? The one on the Sugar Labs website?

I'm amazed at your question.  What other content could I be referring to?  Content is information about Sugar Labs, our products, our projects, our events, and our people.  Themes or stylesheets is the implementation of the arrangement of the content.

As fewer and fewer people _do anything_ at Sugar Labs, our content flow necessarily shrinks, and the content that is hosted on the site ages and reduces in relevance.  This is not a technical problem, but a social one.  It cannot be solved by technical means.

**Comment by amaaniqbal** [comment]
> We don't need a workflow to solve problems.

I meant have you thought of any priorities of solving problems. In the sense, what should be targeted first and to what extent or which all tasks are required to be done ASAP, etc?

> What other content could I be referring to?

:sweat_smile: I thought you could be referring to some regular content writing for newsletters or something else apart from that on our website which is directly presented to our users. Sorry!

**Comment by quozl** [comment]
> I meant have you thought of any priorities of solving problems. In the sense, what should be targeted first and to what extent or which all tasks are required to be done ASAP, etc?

No.

Thinking about in, I do notice one critical absence of content.  We're a software project, but we don't talk about what we do now.

Developers who make releases of Sugarizer, Music Blocks, Sugar, and activities could add to their checklists the writing of an announcement on the web site as well as to the mailing lists.  That hasn't been happening.  We don't have a process.  There's a blog link on the site, but it is to an aggregator and several of the people listed don't work with Sugar Labs any longer.

**Comment by amaaniqbal** [comment]
> We're a software project, but we don't talk about what we do now.

But I guess it's important for any newcomer to know what is actually required to be done apart from fixing borders here and there.

I see a way to revamp some activity related content during GCI '19 via some Quality Assurance or Research related tasks. Core content like Mission, About and related content would not change much I guess.

So how should we go about this issue? I see some silly issues related to search bar and responsiveness regularly due to some obsolete frameworks. What are your views on it except hardcoding some lines to fix a specific issue which results in another one sometime later?

**Comment by quozl** [comment]
> But I guess it's important for any newcomer to know what is actually required to be done apart from fixing borders here and there.

No, I disagree.  Any newcomer should be able to figure out for themselves what needs doing, just by trying to use our software and thinking for themselves.  See [How to get started as a Sugar Labs developer](http://lists.sugarlabs.org/archive/sugar-devel/2019-September/057182.html) in total, and especially the section "How to ask for directions?"

> I see a way to revamp some activity related content during GCI '19 via some Quality Assurance or Research related tasks. Core content like Mission, About and related content would not change much I guess.

No, I disagree again.  Students in Google Code-in know so little about the context of our project and the activities that the activity documentation they've provided is of very poor quality, and we haven't had the reviewers to get changes made before the tasks run out of time.  This could only change if we had mentors who themselves used the activities, and very few of the mentors on offer have ever got so involved.  If they had, we'd have seen bug reports and changes.

> So how should we go about this issue? I see some silly issues related to search bar and responsiveness regularly due to some obsolete frameworks. What are your views on it except hardcoding some lines to fix a specific issue which results in another one sometime later?

I don't know how to solve this issue, but I will continue to resist noise without signal; like changes to style and appearance without changes to content.

**Comment by amaaniqbal** [comment]
> See How to get started as a Sugar Labs developer in total

I have already gone through this way back. But has missed "How to ask for directions?" section 😅

> This could only change if we had mentors who themselves used the activities

I had used Music Blocks way back but couldn't stick with it.

> This could only change if we had mentors who themselves used the activities, and very few of the mentors on offer have ever got so involved. If they had, we'd have seen bug reports and changes.

Okay, I will try to dig deeper into any of Sugar, Sugarizer or Music Blocks soon. Mostly before the beginning of next month. But I cannot guarantee this month.

> I don't know how to solve this issue, but I will continue to resist noise without signal; like changes to style and appearance without changes to content.

So let's wait to see what opinion others have on this.

**Comment by Praharx** [comment]
@quozl  Can I take this up is this open right now?

**Comment by quozl** [comment]
Thanks for your offer, but I don't see any contributions from you to our software; Sugarizer, Music Blocks, Sugar, or activities.  I'm not sure you know what you are in for.

That said, a technical improvement to the files in the repository would be to;
* choose a well-known well-used theme from an upstream project, (which one, and why?),
* remove all customisation of the existing theme,
* change to the chosen theme,
* ensure the content is properly tagged without customisation other than choice of style,
* present the content for us to review.

Are you sure you want to try?  You might read through the other issues and pull requests.  Remember also the Sugar Labs software projects are now less frequently maintained than previously, and almost all of the content on the site is now out of date.  So you'd be spending time tidying up the web site only to find that the content is wrong, unmaintained, and you have nobody wanting to write new content.  By content I mean things to say to visitors, not the style or appearance of the things to say.  A more rational approach would be to delete out of date content _before_ starting any work.  And to do that you need to be fully aware of what Sugar Labs has done and is doing.  That's why I began by questioning why you.  Hope that helps!
